tests/test_error_handling.py

```diff
@@ -0,0 +1,88 @@
+from unittest.mock import MagicMock
+import inspect
+import random
+import string
+
+from pytest import fixture
+
+import senza.error_handling
+
+
+def lineno() -> int:
+    """
+    Returns the current line number.
+    Needed to make sure the expected exceptions stack traces are always updated
+    """
+    return inspect.currentframe().f_back.f_lineno
+
+
+def generate_fake_filename() -> str:
+    return ''.join(random.choice(string.ascii_uppercase + string.digits)
+                   for _ in range(10))
+
+@fixture()
+def mock_tempfile(monkeypatch):
+    random_name = generate_fake_filename()
+    mock = MagicMock(name=random_name)
+
+    mock.__enter__.return_value = mock
+    mock.return_value = mock
+    monkeypatch.setattr('senza.error_handling.NamedTemporaryFile', mock)
+    return mock
+
+
+def test_store_exception(monkeypatch, mock_tempfile):
+
+    line_n = lineno() + 2
+    try:
+        _ = 0/0
```

Comment:
Maybe `raise Exception("Testing exception handing")` would do better here than an elaborated code to force an exception?